rskj-core/src/main/java/co/rsk/rpc/modules/eth/EthModule.java

```diff
@@ -150,15 +151,14 @@ public String call(CallArguments args, String bnOrId) {
         }
     }
 
-    public String estimateGas(CallArguments args) {
+    public String estimateGas(CallArguments args, @Nonnull Block block) {
```

Comment:
in general logic looks good to me, but the approach used doesn't look consistent with other module classes. Eg. `co.rsk.rpc.modules.eth.EthModule#call(CallArguments args, String bnOrId)` accepts "raw" string `bnOrId` value, or `co.rsk.rpc.modules.eth.EthModuleWallet#sign(String addr, String data)` also accepts `addr` in raw format. Parsing itself happens in a specific module implementation class.

And in your case the work is being done in the "facade" class - `Web3Impl`

Reply:
It has sense, code was updated!